internal/rootcoord/create_collection_task.go

```diff
@@ -304,11 +312,10 @@ func (t *createCollectionTask) Execute(ctx context.Context) error {
 		return nil
 	}
 
-	// check collection number quota for the entire the instance
-	existedCollInfos, err := t.core.meta.ListCollections(ctx, typeutil.MaxTimestamp)
+	existedCollInfos, err := t.core.meta.ListCollections(ctx, t.Req.GetDbName(), typeutil.MaxTimestamp, true)
```

Comment:
Better to push the count logic down to meta table, otherwise a lot of copy of collection meta will be allocated.